### ISSUE TITLE
Messages from application stops working

### ISSUE THREAD

**Issue by Szarp** [AllanWang/Frost-for-Facebook#1732]
- [x] I have verified that the bug does not occur in my mobile browser (at https://touch.facebook.com/)
- [x] The bug occurs at mobile browser

**Describe the bug**
Messeges from Facebook stop working. As I noticed, they disabled page: https://m.facebook.com/messages/ and no messages are visible

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to 'messeges'
2. See error

**Details (please provide at least the app version):**
 - App Version: 2.4.7
 - Device: Hauwei P8 lite
 - Android Version: 8.0.0

**Comment by Ansem93** [comment]
Same issue with Sony XZ1 with Android 9.
On touch.facebook.com I recieve this error: "it's possible that the link is not correct or the page has been removed"

**Comment by dpriskorn** [comment]
I have the same issue. I guess FB finally killed the old way of accessing messages. In icecat on my mobile with lineageos both touch and m. addresses return the error described above.

**Comment by creativetrendsapps** [comment]
Do messages work on the basic site? https://mbasic.facebook.com/messages

**Comment by dpriskorn** [comment]
No, the same

**Comment by creativetrendsapps** [comment]
> No, the same

Are you in the US?

**Comment by dpriskorn** [comment]
No Sweden. Swedish IP

**Comment by creativetrendsapps** [comment]
> No Sweden. Swedish IP

I figured you weren't. I think this is a regional bug on Facebook's end.

**Comment by dpriskorn** [comment]
A ha. OK, I'm gonna repot it on the feedback form.

**Comment by dpriskorn** [comment]
They recently changed the ad manager also for European customers, which resulted in errors, so it might be related.

**Comment by Szarp** [comment]
It might be related with Messneger app crash:
Link to news:
https://www.google.com/search?q=messenger+app+crash&client=firefox-b-e&source=lnms&tbm=nws&sa=X&ved=2ahUKEwjSqJWkkcbtAhWn-ioKHfLqBSgQ_AUoAXoECBsQAw&biw=1516&bih=788

**Comment by Ansem93** [comment]
messenger.com still works, but you need to use the desktop view :/

**Comment by dpriskorn** [comment]
I bought a new phone and installed Messenger Lite and Facebook Business Manager and there I got a warning about messages might not work correctly in Europe due to gdpr related changes.
It works though, so I can talk to my customers. I will have to keep everything private of the phone because I trust neither Google nor Facebook (nor Samsung) running non-free code on my device.

**Comment by freeeflyer** [comment]
Same here.. Messages stopped working like 1 or 2 weeks ago.. (French IP)

**Comment by AThokay** [comment]
Same here... I tried from VPN (Texas IP), but not working.

**Comment by t0ma5** [comment]
same here, European IP... same problem in Frost and Firefox/Chrome running on Windows Desktop. The problem is Facebook and not Frost

**Comment by joergi** [comment]
Also not working, German IP on fairphone 3+ with Android 10  on Frost 2.4.7

**Comment by illtellyoulater** [comment]
Can confirm, messages page not working anymore (European IP, Android 10, latest public build of Frost).

**Comment by vinibali** [comment]
Same here with a Hungarian IP

**Comment by r0d0dendr0n** [comment]
Same here.

Poland can not into messenger.

**Comment by Blackfire-dev** [comment]
Same with OnePlus 7T Pro, Android 10 In the UK.
mbasic.facebook.com; m.facebook.com, touch.facebook.com on browser doesn't work either.

Direct Messages and Page Messages are the same result.

**Comment by t0ma5** [comment]
stop adding reports, IPs, devices, etc///, it's pointless....
it's not FROST, neither your system or your IP... it's FACEBOOK... same happens in desktop browser when accessing the mobile version of Facebook.

**Comment by vegancorr** [comment]
I'm pretty sure this bug is on the Facebook's features list.

**Comment by vinibali** [comment]
> stop adding reports, IPs, devices, etc///, it's pointless....
> it's not FROST, neither your system or your IP... it's FACEBOOK... same happens in desktop browser when accessing the mobile version of Facebook.

i think we all know that. we just keep this topic updated.

**Comment by Blackfire-dev** [comment]
> 
> 
> stop adding reports, IPs, devices, etc///, it's pointless....
> it's not FROST, neither your system or your IP... it's FACEBOOK... same happens in desktop browser when accessing the mobile version of Facebook.

We all know this, we all use Frost and similar applications to avoid facebook tracking and such. Only two options, Either AllanWang does some amazing coding work to circumvent the new Facebook intentional bug (which is the point in these "reports"), or abandons the messaging feature of Frost all together. I personally hope for the former. I posted on this rather than creating a new ticket to remind AllanWang that this app is used in all corners of world on many devices and we are rooting for a resolution.

I'm not apologising for anything if that's what you are after.

**Comment by t0ma5** [comment]
@Blackfire-dev  it looks like Facebook wants us to use Messenger from now on... I am following other alternative apps like Frost (paid and free) and so far no workaround has been implemented. Meanwhile, I'm using Messenger Lite. No apologies expected. I simply thought it must be quite annoying for everyone to get notifications on posts giving IP, country, device details, etc when that is totally unrelated to the problem/situation. Nothing else.

@AllanWang What are the chances of a workaround to keep the messages alive in Frost?

**Comment by AllanWang** [comment]
If Facebook closes off certain mobile pages based on location, I don't think there's much I'm going to do/look into. If there is still a separate user agent that works, maybe I can add that.

In short, this currently does not affect the US. Parsing is also done with mbasic, so maybe notifications will remain functional for those of you affected.

The long term solution would be to add support for messenger.com, but that is a pretty significant task. I'm still busy with work due to EOY deadlines, and I'm not sure how much time I'd conceivably spend coding during my break as we've all felt the effects of quarantine.

Feel free to subscribe to this issue for updates, but yes, device and version info won't be very helpful here.

If you are testing by the way, know that the messages page has never worked with a mobile user agent within the past few years. You need to specify a desktop agent. 

I personally have messenger installed because I need push notifications, but I also know that people want the feature. I don't use 99% of the features in Frost anymore since I basically no longer use Facebook, but I'm still interested in maintaining it given it's one of the few solutions available for free.

**Comment by creativetrendsapps** [comment]
> If Facebook closes off certain mobile pages based on location, I don't think there's much I'm going to do/look into. If there is still a separate user agent that works, maybe I can add that.
> 
> In short, this currently does not affect the US. Parsing is also done with mbasic, so maybe notifications will remain functional for those of you affected.
> 
> The long term solution would be to add support for messenger.com, but that is a pretty significant task. I'm still busy with work due to EOY deadlines, and I'm not sure how much time I'd conceivably spend coding during my break as we've all felt the effects of quarantine.
> 
> Feel free to subscribe to this issue for updates, but yes, device and version info won't be very helpful here.
> 
> If you are testing by the way, know that the messages page has never worked with a mobile user agent within the past few years. You need to specify a desktop agent. 
> 
> I personally have messenger installed because I need push notifications, but I also know that people want the feature. I don't use 99% of the features in Frost anymore since I basically no longer use Facebook, but I'm still interested in maintaining it given it's one of the few solutions available for free.

I don't think they're closing it out. I think they're just experiencing an issue. Maybe try implementing the desktop Messenger as a workaround? Maki has it included as an option. I'm seriously thinking of implementing it as well (as a temporary workaround). Hopefully though, Facebook fixes the issue.

**Comment by creativetrendsapps** [comment]
> @Blackfire-dev  it looks like Facebook wants us to use Messenger from now on... I am following other alternative apps like Frost (paid and free) and so far no workaround has been implemented. Meanwhile, I'm using Messenger Lite. No apologies expected. I simply thought it must be quite annoying for everyone to get notifications on posts giving IP, country, device details, etc when that is totally unrelated to the problem/situation. Nothing else.
> 
> @AllanWang What are the chances of a workaround to keep the messages alive in Frost?

This isn't technically true. Maki (and FaceSlim) have the desktop version of Messenger that are... _Useable_.

**Comment by freeeflyer** [comment]
> This isn't technically true. Maki (and FaceSlim) have the desktop version of Messenger that are... _Useable_.

As for faceslim, messaging function does not work from here (france)..

**Comment by creativetrendsapps** [comment]
> > This isn't technically true. Maki (and FaceSlim) have the desktop version of Messenger that are... _Useable_.
> 
> As for faceslim, messaging function does not work from here (france)..
> 
> 

In FaceSlim you can swipe out the left navigation drawer and use the messenger.com link. It's not in anyway optimized however. If anyone uses Maki, you can choose the "Messenger in Maki" option for messaging clients.

**Comment by AllanWang** [comment]
None of the apps I've seen have implemented Messenger in a way that I'd like, but it's also really not mobile friendly. I'm sure we all know how to do it to some extent, but it'll take time.

**Comment by creativetrendsapps** [comment]
> None of the apps I've seen have implemented Messenger in a way that I'd like, but it's also really not mobile friendly. I'm sure we all know how to do it to some extent, but it'll take time.

Yes you're correct. I do think Maki has the best implemention thus far though. Best of luck.

**Comment by t0ma5** [comment]
@creativetrendsapps where can I find that FaceSlim you talk about? The one I know made its last release in 2018

> In short, this currently does not affect the US

@AllanWang I tried connecting with 4 different US VPNs but didn't change anything for me so it's not IP related but the server where ur profile is?

**Comment by creativetrendsapps** [comment]
> @creativetrendsapps where can I find that FaceSlim you talk about? The one I know made its last release in 2018
> 
> > In short, this currently does not affect the US
> 
> @AllanWang I tried connecting with 4 different US VPNs but didn't change anything for me so it's not IP related but the server where ur profile is?

It's on GitHub as well as F-Droid.

I don't think the issue is IP related. Though I think it's strange that users in the UK, Germany and other regions other than the US are experiencing this. 🤔 Could it be a cookie issue? Has anyone tried logging out logging back in and accepting cookies?

**Comment by creativetrendsapps** [comment]
Actually I think this may be either regional OR account based. Facebook may actually be removing messages from their site after all. I say that because I used a vpn and also faked my location. I can still access messages though. 🤷🏾‍♂️

**Comment by t0ma5** [comment]
> It's on GitHub as well as F-Droid.

that's the one I know (I used it for a while) but the latest release is from **2018**, github and f-droid 
https://github.com/indywidualny/FaceSlim/releases 
https://f-droid.org/packages/org.indywidualni.fblite/

**Comment by creativetrendsapps** [comment]
> > It's on GitHub as well as F-Droid.
> 
> that's the one I know (I used it for a while) but the latest release is from **2018**, github and f-droid 
> https://github.com/indywidualny/FaceSlim/releases 
> https://f-droid.org/packages/org.indywidualni.fblite/

I don't think he'll be updating it anymore. 

https://github.com/indywidualny/FaceSlim/issues/411

Though messenger is still implemented there. Just not good.

**Comment by t0ma5** [comment]
yeah, dead end, dev just posted "Messenger lite is not bad in fact IMHO. I use a normal one anyway. Just disabled all its permissions." @creativetrendsapps

**Comment by creativetrendsapps** [comment]
> yeah, dead end, dev just posted "Messenger lite is not bad in fact IMHO. I use a normal one anyway. Just disabled all its permissions." @creativetrendsapps
> 

Messenger Lite isn't bad though. I admittedly use it as well, 🤷🏾‍♂️.

**Comment by t0ma5** [comment]
> Messenger Lite isn't bad though. I immediately use it as well, 🤷🏾‍♂️.

same here.. dark mode wouldn't hurt though ;)

**Comment by r0d0dendr0n** [comment]
Fk that, installing fb apps as an override is unacceptable in my opinion. I would rather give up fb than install proprietary fb code. For communication I use signal anyway. I simply use a web browser woth desktop mode once a day.

Maybe Diaspora anyone?

**Comment by ghost** [comment]
I presume this will not help as the problem is pretty clear but just in case...
Messenger is still accessible on [WebApps](https://github.com/tobykurien/WebApps/blob/HEAD/README.md) using a desktop user agent.
As it is open source and up to date, maybe using a part of their code can simplify the task.

**Comment by creativetrendsapps** [comment]
> I presume this will not help as the problem is pretty clear but just in case...
> Messenger is still accessible on [WebApps](https://github.com/tobykurien/WebApps/blob/HEAD/README.md) using a desktop user agent.
> As it is open source and up to date, maybe using a part of their code can simplify the task.

🤔 So it works in that app and still doesn't in Frost? Is there a desktop setting in that app?

**Comment by creativetrendsapps** [comment]
> I presume this will not help as the problem is pretty clear but just in case...
> Messenger is still accessible on [WebApps](https://github.com/tobykurien/WebApps/blob/HEAD/README.md) using a desktop user agent.
> As it is open source and up to date, maybe using a part of their code can simplify the task.

Messenger or the mobile version of messages?

**Comment by ghost** [comment]
> > I presume this will not help as the problem is pretty clear but just in case...
> > Messenger is still accessible on [WebApps](https://github.com/tobykurien/WebApps/blob/HEAD/README.md) using a desktop user agent.
> > As it is open source and up to date, maybe using a part of their code can simplify the task.
> 
> Messenger or the mobile version of messages?

The web page messenger.com, but I have no idea how hard it can be to "convert" the result to a cleaner, more adapted design for mobile.

**Comment by creativetrendsapps** [comment]
> > > I presume this will not help as the problem is pretty clear but just in case...
> > > Messenger is still accessible on [WebApps](https://github.com/tobykurien/WebApps/blob/HEAD/README.md) using a desktop user agent.
> > > As it is open source and up to date, maybe using a part of their code can simplify the task.
> > 
> > Messenger or the mobile version of messages?
> 
> The web page messenger.com, but I have no idea how hard it can be to "convert" the result to a cleaner, more adapted design for mobile.

Oh. Well, yeah. Applying a desktop user agent would load the messenger.com site. Making it look good on a mobile device would just involve some CSS.

**Comment by eapo** [comment]
> > > > I presume this will not help as the problem is pretty clear but just in case...
> > > > Messenger is still accessible on [WebApps](https://github.com/tobykurien/WebApps/blob/HEAD/README.md) using a desktop user agent.
> > > > As it is open source and up to date, maybe using a part of their code can simplify the task.
> > > 
> > > 
> > > Messenger or the mobile version of messages?
> > 
> > 
> > The web page messenger.com, but I have no idea how hard it can be to "convert" the result to a cleaner, more adapted design for mobile.
> 
> Oh. Well, yeah. Applying a desktop user agent would load the messenger.com site. Making it look good on a mobile device would just involve some CSS.

Looks like a solution, but instead of CSS, and talking about this here I just started a new topic there: [www.facebook.com (the desktop site) doesn't work · Issue #222 · tobykurien/WebApps](https://github.com/tobykurien/WebApps/issues/222#issuecomment-746078473)

**Comment by eapo** [comment]
Would be a good **solution** to add
1. **_User agent switcher_** and 
2. **a new customizable _Toggle script/CSS_ menu element** in the *pull menu* 
    - with a predefined script like `document.getElementsByClassName('n5hh3s10')[0].style.display = "none";` and `document.getElementsByClassName('n5hh3s10')[0].style.display = "block";`
3. to be able to use the _messenger dot com_'s desktop version in the app.
**Who can realize that?**
Please 👍!

**Comment by akorn** [comment]
> Messenger is still accessible on [WebApps](https://github.com/tobykurien/WebApps/blob/HEAD/README.md) using a desktop user agent.

Unfortunately no; see https://github.com/tobykurien/WebApps/issues/221

**Comment by ghost** [comment]
> > Messenger is still accessible on [WebApps](https://github.com/tobykurien/WebApps/blob/HEAD/README.md) using a desktop user agent.
> 
> Unfortunately no; see [tobykurien/WebApps#221](https://github.com/tobykurien/WebApps/issues/221)

Strange, the website messenger.com works perfectly for me with a desktop user agent (if we disregard the fact that the interface is bad on a phone).
Can anyone else try?

**Comment by Ansem93** [comment]
I found an app on Android "bluer for facebook". where messenger actually works. I think that they're using messenger.com and not touch.facebook. 
Also I can confirm that messenger.com using a desktop user agent works in Samsung Browser.

**Comment by t0ma5** [comment]
@Ansem93 I'll give it a try. thanks! https://thebluer.app/

**Comment by creativetrendsapps** [comment]
> I found an app on Android "bluer for facebook". where messenger actually works. I think that they're using messenger.com and not touch.facebook. 
> Also I can confirm that messenger.com using a desktop user agent works in Samsung Browser.

Bluer is using Flutter to display all of the content in the app. I'm not sure if they're actually using the messenger.com site for messages however.

Any browser that uses a desktop user agent will display messenger.com. Making it look good and function is different.

**Comment by Ansem93** [comment]
I just wanted to point out an example of a working app with messenger.com where it actually looks good and is usable. 
They surely use some custom css or something like that because even the home page looks different.

Actually, from the aesthetics point of view, the app is great. It's a pity that's closed :/

**Comment by t0ma5** [comment]
@Ansem93 BLUER works great for messages! similar to the official Messenger Lite but with dark theme. 
Thanks once again for the tip! It serves well as a temporary solution

**Comment by eapo** [comment]
> 
> 
> > > Messenger is still accessible on [WebApps](https://github.com/tobykurien/WebApps/blob/HEAD/README.md) using a desktop user agent.
> > 
> > 
> > Unfortunately no; see [tobykurien/WebApps#221](https://github.com/tobykurien/WebApps/issues/221)
> 
> Strange, the website messenger.com works perfectly for me with a desktop user agent (if we disregard the fact that the interface is bad on a phone).
> Can anyone else try?

Useful on large screens, but useless on normal screens in portrait mode. Here is an example from the [Unfinished Messages to Messenger hack and return to "touch.$FACEBOOK_COM" by eapo · Pull Request #1734 · AllanWang/Frost-for-Facebook](https://github.com/AllanWang/Frost-for-Facebook/pull/1734)
![the Unfinished Messages to Messenger hack](https://i.imgur.com/azOFGlj.png)

**Comment by eapo** [comment]
> @Ansem93 I'll give it a try. thanks! https://thebluer.app/

How can you trust in them?

**Comment by t0ma5** [comment]
> 
> 
> > @Ansem93 I'll give it a try. thanks! https://thebluer.app/
> 
> How can you trust in them?

Can I trust any closed source app from the Play Store? Can I trust Google? Can I trust my phone's manufacturer? Can I trust my ISP? No idea... Even @AllanWang looks quite suspicious to me :P

**Comment by Ansem93** [comment]
> 
> 
> > @Ansem93 I'll give it a try. thanks! https://thebluer.app/
> 
> How can you trust in them?

I trust them more than facebook. Still, I didn't want to suggest the app. My intention was to show a working implementation of  messenger. com inside an app.

**Comment by creativetrendsapps** [comment]
> > 
> > 
> > > @Ansem93 I'll give it a try. thanks! https://thebluer.app/
> > 
> > How can you trust in them?
> 
> I trust them more than facebook. Still, I didn't want to suggest the app. My intention was to show a working implementation of  messenger. com inside an app.
> 

🤦🏾‍♂️ It's not messenger.com.

**Comment by AllanWang** [comment]
> Can I trust any closed source app from the Play Store? Can I trust Google? Can I trust my phone's manufacturer? Can I trust my ISP? No idea... Even @AllanWang looks quite suspicious to me :P

That is true, but I'd also want to emphasize that you should not blindly trust third party web wrappers, including mine. Sure, if there are enough users and there's an actual privacy policy, maybe it's expected to be safe, but you're still giving us your passwords, and not some kind of access token. Mistakes also do happen; I've accidentally logged cookie ids before in the early stages of Frost. Even without malicious intent, you are assuming your own risks by using applications like these.

**Comment by infeeeee** [comment]
I got this notification on desktop messenger.com today, so I guess facebook is aware of the issue:
![kép](https://user-images.githubusercontent.com/1271737/102398374-7ea60b00-3fdf-11eb-8961-37c2f72152f4.png)

**Comment by AllanWang** [comment]
> Would be a good **solution** to add
> 
> 1. **_User agent switcher_** and

This is what got Frost banned btw. I used different user agents to help with different features, and now it's back to desktop only after the phishing warnings. Maybe it's fine if you switch the user agent once and never again after login, but people are going to complain if an agent switcher logs them out.

> 2. **a new customizable _Toggle script/CSS_ menu element** in the _pull menu_
>    
>    * with a predefined script like `document.getElementsByClassName('n5hh3s10')[0].style.display = "none";` and `document.getElementsByClassName('n5hh3s10')[0].style.display = "block";`

What are you expecting users to do with this? I get that it's more customization, but you're going to need to go on your desktop just to inspect elements. If you want to style things yourself, Frost is open source and you can just make your own version. If it's something that would be useful to the community, it would be worth vetting and adding as a release for everyone.

> 3. to be able to use the _messenger dot com_'s desktop version in the app.
>    **Who can realize that?**
>    Please 👍!

It's been realized (poorly), but it's also messenger's fault for hardcoding so many sizes. I'm not sure if we can do much better without having a super brittle implementation.

**Comment by eapo** [comment]
> I get that it's more customization, but you're going to need to go on your desktop just to inspect elements. If you want to style things yourself,

Sorry, I wasn't clear enough. Hiding the `div.n5hh3s10` will hide the [_profile_ panel from the right side of the messenger's window](#issuecomment-747091485 "see below") to allow users with normal and smaller screen sizes to use messenger's desktop version.

![](https://i.imgur.com/azOFGlj.png)

**Comment by t0ma5** [comment]
> 
> 
> I got this notification on desktop messenger.com today, so I guess facebook is aware of the issue:
> ![kép](https://user-images.githubusercontent.com/1271737/102398374-7ea60b00-3fdf-11eb-8961-37c2f72152f4.png)

me too, there is hope :)
more info here: https://mspoweruser.com/facebook-messenger-says-some-features-not-available-heres-why/

**Comment by KopiasCsaba** [comment]
Yep, hello from Hungary too:-)

By the way thank you very very much for this app. It's a life saver.

**Comment by KopiasCsaba** [comment]
Hmm...
![image](https://user-images.githubusercontent.com/7661267/102543698-fa718780-40b3-11eb-8c21-1b3e5d58ae29.png)

**Comment by eapo** [comment]
> > yeah, dead end, dev just posted "Messenger lite is not bad in fact IMHO. I use a normal one anyway. Just disabled all its permissions." @creativetrendsapps
> 
> Messenger Lite isn't bad though. I admittedly use it as well, 🤷🏾‍♂️.

Same here, but with disabled services like these:
https://hackmd.io/@eaposztrof/Facebook-Messenger-Secure-OpenSource

**Comment by creativetrendsapps** [comment]
The issue is GDPR related.

**Comment by t0ma5** [comment]
@eapo Gold! thanks for the info, I was just looking for an open-source replacement for App Ops by github.com/RikkaApps/

> The issue is GDPR related.

No doubt about it, as I posted yesterday:

> more info here: https://mspoweruser.com/facebook-messenger-says-some-features-not-available-heres-why/

**Comment by t0ma5** [comment]
Friendly for Facebook (wrapper) also came out with a nice work-around for the time being (for EU users).
Screenshots: 
https://i.imgur.com/NFzOCix.jpg
https://i.imgur.com/qA6Wn21.jpg

**Comment by creativetrendsapps** [comment]
Just throwing it out there... This works for me and the dev of Maki:

This works for me:

/CSS/

.wkznzc2l {
display: none !important;
}

.kuivcneq {
display: none !important;
}

.bafdgad4 {
display: none !important;
}

.aov4n071.cxmmr5t8.bi6gxh9e.hcukyx3x.jb3vyjys.hv4rvrfc.qt6c0cv9.dati1w0a {
display: none !important;
}

.rj1gh0hx {
max-width: -webkit-fill-available;
}

.j83agx80 {
max-width: -webkit-fill-available;
}

/web settings to make it look better, play with the scale a bit for your needs/
~~mWebView.setInitialScale(160);~~ <--don't use.

![photo_2020-12-24_16-27-10](https://user-images.githubusercontent.com/14278089/103106678-782e1980-4605-11eb-9da9-99df184c7dc2.jpg)

**Comment by sfilmak** [comment]
I am not sure it will help, but it is worth trying reporting this bug directly to FB (just like I did): https://www.facebook.com/business/help/1669772506455012?id=858953797905263
Do not forget to attach a screenshot (preferably from Chrome or any other browser).

You won't be able to do it from the mobile web version (as far as I can see), so you should use an official app or desktop version.

**Comment by riku-fr** [comment]
Dont know if that helps, but i can access it from my browser (Vivaldi mobile), i can access the integrated messenger by using the computer agent (use computer version)

**Comment by creativetrendsapps** [comment]
> Dont know if that helps, but i can access it from my browser (Vivaldi mobile), i can access the integrated messenger by using the computer agent (use computer version)

Mostly likely because you're not in the EU. If you can access messages in your browser... You can access it in Frost or any other wrapper.

**Comment by riku-fr** [comment]
i'm in France, but what i am saying has already been posted here when i read the entire thread.
Just to precise anyway, opening https://facebook.com/ with the default agent of my mobile and then going in messenger displays the content doesn't exist, same result if i try to open https://facebook.com/messages/

If i open https://facebook.com/ as a desktop agent and then go on Messenger it works asking me to open the page with my browser, note that the url is now different, i guess this is a personal identifier, this is now https://facebook.com/messages/t/xxxxxxxxxxxxxx where x is a numeric value

**Comment by creativetrendsapps** [comment]
> i'm in France, but what i am saying has already been posted here when i read the entire thread.
> Just to precise anyway, opening https://facebook.com/ with the default agent of my mobile and then going in messenger displays the content doesn't exist, same result if i try to open https://facebook.com/messages/
> 
> If i open https://facebook.com/ as a desktop agent and then go on Messenger it works asking me to open the page with my browser, note that the url is now different, i guess this is a personal identifier, this is now https://facebook.com/messages/t/xxxxxxxxxxxxxx where x is a numeric value

What you're talking about is completely different from what the issue is though. 

Yes the Facebook site works with messages as it always has. 

The Facebook mobile site doesn't.

Going to https://facebook.com is different than https://m.facebook.com or https://touch.facebook.com.

**Comment by gothmog123** [comment]
Sorry can someone give me the skinny on this? I mean is there any chance its gonna be fixed in the long run? I really hate using messenger lite

**Comment by lokomaster** [comment]
Just saying there is working alternative called Slimsocial on Fdroid. Maybe not absolutely perfect working, but only oné working opensource alternative

**Comment by creativetrendsapps** [comment]
> Just saying there is working alternative called Slimsocial on Fdroid. Maybe not absolutely perfect working, but only oné working opensource alternative 

Not technically. FaceSlim is also open source and is also (just like most the other apps with workarounds), using messenger.com

**Comment by lokomaster** [comment]
> > Just saying there is working alternative called Slimsocial on Fdroid. Maybe not absolutely perfect working, but only oné working opensource alternative
> 
> Not technically. FaceSlim is also open source and is also (just like most the other apps with workarounds), using messenger.com

Sure, but is there is better option now, when Frost is broken? I think it is still better, than use data gathering official App, or proprietal apps recommended above.

**Comment by creativetrendsapps** [comment]
> > > Just saying there is working alternative called Slimsocial on Fdroid. Maybe not absolutely perfect working, but only oné working opensource alternative
> > 
> > Not technically. FaceSlim is also open source and is also (just like most the other apps with workarounds), using messenger.com
> 
> Sure, but is there is better option now, when Frost is broken? I think it is still better, than use data gathering official App, or proprietal apps recommended above. 

While off topic from the original issue.

I'm sure Facebook is still gathering your data even while using a wrapper 🤷🏾‍♂️.

**Comment by AllanWang** [comment]
No guarantees, but I will try to look at this during the weekend.

For those where /messages doesn't work, I take it that notifications don't work either? Parsing from messenger.com is different, and I believe they have completely different sets of ids.

**Comment by rochr4** [comment]
Pretty broken, pls 'fix".
![Screenshot_20210106-233903_Frost](https://user-images.githubusercontent.com/60553713/103827080-75f1a380-5078-11eb-95e5-11776df15f88.jpg)

**Comment by kyelbek** [comment]
> For those where /messages doesn't work, I take it that notifications don't work either? Parsing from messenger.com is different, and I believe they have completely different sets of ids.

Sadly, they're not. At first when they blocked it for us, I was able to see messages notifications without any problems, also badge on messaging icon was there. Not anymore though.
Notifications for messages aren't that important in my opinion, so I wouldn't bother for now if it's gonna be problematic for you. Let's be honest, if someone is after reliability because of how important messenger is in his life, they should go back to Messenger.
It's nice enough from you that you want to look into it, even if there's nothing wrong with current implementation in where you live.

Cheers!

**Comment by mormegil-cz** [comment]
Apparently, the messages (under `/messages`) work only in the non-mobile web: if on a mobile phone I click on the Messenger link in the hamburger menu, the hurt-thumb error page appears; if I switch the browser to desktop mode, the link redirects to the desktop version under www.facebook.com which works fine.

(And yes, the unread-messages indicator badge used to work for some time but it does not anymore. The mobile web does not show it, either. Only the desktop web does.)

**Comment by AllanWang** [comment]
First test apk for those who want to try: https://github.com/AllanWang/Frost-for-Facebook-APK-Builder/releases/tag/v2884

I didn't test this specific build, but I've been testing the debug ones.

Install the update, go to `settings > appearance > main activity tabs`, and add the messenger tab.

What isn't added:
* Multi account support
* Login credential saves
* Parsing (notifications, unread badge)

Notes:
* Themes are supported; currently needs to be opaque based on how messenger works
* Sending messages seems to work, but it's slow and you need to press enter for the emoji to turn into the send icon (or close your keyboard). Not sure if other apps have the same problem.

All in all, pretty bad user experience, but it works.

**Comment by Pyro1992** [comment]
> First test apk for those who want to try: https://github.com/AllanWang/Frost-for-Facebook-APK-Builder/releases/tag/v2884
> 
> I didn't test this specific build, but I've been testing the debug ones.
> 
> Install the update, go to `settings > appearance > main activity tabs`, and add the messenger tab.
> 
> What isn't added:
> 
> * Multi account support
> * Login credential saves
> * Parsing (notifications, unread badge)
> 
> Notes:
> 
> * Themes are supported; currently needs to be opaque based on how messenger works
> * Sending messages seems to work, but it's slow and you need to press enter for the emoji to turn into the send icon (or close your keyboard). Not sure if other apps have the same problem.
> 
> All in all, pretty bad user experience, but it works.

Can Confirm it works with latest test build

**Comment by freeeflyer** [comment]
> > First test apk for those who want to try: https://github.com/AllanWang/Frost-for-Facebook-APK-Builder/releases/tag/v2884

Thanks for the release.. 
Just installed it, here is my experience. (from RGPD enabled France)

> Can Confirm it works with latest test build
I had to re-login on the messenger tab (maybe it's a cookie thing). It kinda work.. 
but on my phone, it's nearly unusable.. 
Here is a vertical version :
![Screenshot_20210113-105242](https://user-images.githubusercontent.com/12149490/104438142-f3914400-558f-11eb-9548-061a80ea5762.png)
As you see the contact description is masking the message section..

Here is an horizontal version:
![Screenshot_20210113-105252](https://user-images.githubusercontent.com/12149490/104438154-f68c3480-558f-11eb-8900-673d446cdabb.png)
It's not very usable either when the keyboard pops up..

Thanks again.

**Comment by Pyro1992** [comment]
> > > First test apk for those who want to try: https://github.com/AllanWang/Frost-for-Facebook-APK-Builder/releases/tag/v2884
> 
> Thanks for the release..
> Just installed it, here is my experience. (from RGPD enabled France)
> 
> > Can Confirm it works with latest test build
> > I had to re-login on the messenger tab (maybe it's a cookie thing). It kinda work..
> > but on my phone, it's nearly unusable..
> > Here is a vertical version :
> > ![Screenshot_20210113-105242](https://user-images.githubusercontent.com/12149490/104438142-f3914400-558f-11eb-9548-061a80ea5762.png)
> > As you see the contact description is masking the message section..
> 
> Here is an horizontal version:
> ![Screenshot_20210113-105252](https://user-images.githubusercontent.com/12149490/104438154-f68c3480-558f-11eb-8900-673d446cdabb.png)
> It's not very usable either when the keyboard pops up..
> 
> Thanks again.

Had the same first. After force close Frost and OPEN it again all worked
![Screenshot_20210113-112018_Frost Test](https://user-images.githubusercontent.com/18252895/104439559-82eb2700-5591-11eb-8139-c7ac50a2ffb6.jpg)

**Comment by creativetrendsapps** [comment]
> > > > First test apk for those who want to try: https://github.com/AllanWang/Frost-for-Facebook-APK-Builder/releases/tag/v2884
> > 
> > Thanks for the release..
> > Just installed it, here is my experience. (from RGPD enabled France)
> > 
> > > Can Confirm it works with latest test build
> > > I had to re-login on the messenger tab (maybe it's a cookie thing). It kinda work..
> > > but on my phone, it's nearly unusable..
> > > Here is a vertical version :
> > > ![Screenshot_20210113-105242](https://user-images.githubusercontent.com/12149490/104438142-f3914400-558f-11eb-9548-061a80ea5762.png)
> > > As you see the contact description is masking the message section..
> > 
> > Here is an horizontal version:
> > ![Screenshot_20210113-105252](https://user-images.githubusercontent.com/12149490/104438154-f68c3480-558f-11eb-8900-673d446cdabb.png)
> > It's not very usable either when the keyboard pops up..
> > 
> > Thanks again.
> 
> Had the same first. After force close Frost and OPEN it again all worked
> ![Screenshot_20210113-112018_Frost Test](https://user-images.githubusercontent.com/18252895/104439559-82eb2700-5591-11eb-8139-c7ac50a2ffb6.jpg)
> 

🙄 Press the info button to remove the panel...

**Comment by creativetrendsapps** [comment]
> > > First test apk for those who want to try: https://github.com/AllanWang/Frost-for-Facebook-APK-Builder/releases/tag/v2884
> 
> Thanks for the release.. 
> Just installed it, here is my experience. (from RGPD enabled France)
> 
> > Can Confirm it works with latest test build
> I had to re-login on the messenger tab (maybe it's a cookie thing). It kinda work.. 
> but on my phone, it's nearly unusable.. 
> Here is a vertical version :
> ![Screenshot_20210113-105242](https://user-images.githubusercontent.com/12149490/104438142-f3914400-558f-11eb-9548-061a80ea5762.png)
> As you see the contact description is masking the message section..
> 
> Here is an horizontal version:
> ![Screenshot_20210113-105252](https://user-images.githubusercontent.com/12149490/104438154-f68c3480-558f-11eb-8900-673d446cdabb.png)
> It's not very usable either when the keyboard pops up..
> 
> Thanks again.
> 
> 
> 
> 

Press the info button to remove the panel....

**Comment by vegancorr** [comment]
I installed the new apk "Frost Test". It didn't update, it installed alongside.

I tried to change the order of the main activity tabs, but it didn't save. The old chat displays the same error, nothing is changed. I was not able to use the new messenger tab.

**Comment by AllanWang** [comment]
Force closing shouldn't matter. For all of you using default themes, none of the css is applied which is why it's like that. Switch to another theme for now. This is an oversight because Facebook has no css for the default theme.

Frost Test is supposed to install alongside debug and release builds. This is intentional. As for the tab saving error, not sure why that's happening.

**Comment by AllanWang** [comment]
For those of you who've tried other clients, is the messaging process the same? Does the send button really not react until you close the keyboard or press enter? I tested on an older device so maybe it's better for the newer ones

**Comment by alekksander** [comment]
It works on my device as slow and irritating as on slimsocial and maki. Maybe a little bit faster on frost, maybe it's only auto—suggestion. 
Frost won't load messagner if opening in external apps is enabled (prompt what to open with).

**Comment by freeeflyer** [comment]
it works for me in dark mode.
Thanks for the fix.

**Comment by ErebosGR** [comment]
> It works on my device as slow and irritating as on slimsocial and maki. Maybe a little bit faster on frost, maybe it's only auto—suggestion.
> Frost won't load messagner if opening in external apps is enabled (prompt what to open with).

Friendly had a recent update that loads the messenger very fast.

**Comment by kyelbek** [comment]
I doubt that it's even possible to make it quick and resources friendly. FB/Messenger sites are pure garbage and wrappers can't fix their issues, sadly.

**Comment by zerodogg** [comment]
@AllanWang I just tried [v2886](https://github.com/AllanWang/Frost-for-Facebook-APK-Builder/releases/tag/v2886). I can add the new messenger tab fine, and it presents me with the login form. However, when I reach the two-factor component of the login process it says to "Please verify your Facebook Account". In a normal browser clicking on the button to continue in this screen redirects to facebook to verify the two factor (just clicking on verify if already logged in). However, in frost it opens an overlay with "The link is invalid or expired". I've tried twice now, it broke the same way both times.

**Comment by kyelbek** [comment]
> @AllanWang I just tried [v2886](https://github.com/AllanWang/Frost-for-Facebook-APK-Builder/releases/tag/v2886). I can add the new messenger tab fine, and it presents me with the login form. However, when I reach the two-factor component of the login process it says to "Please verify your Facebook Account". In a normal browser clicking on the button to continue in this screen redirects to facebook to verify the two factor (just clicking on verify if already logged in). However, in frost it opens an overlay with "The link is invalid or expired". I've tried twice now, it broke the same way both times.

Go to FB on your PC and confirm from notifications, tick remember this device too. After that relogin into messenger on Frost. Should be fine.

**Comment by zerodogg** [comment]
> Go to FB on your PC and confirm from notifications, tick remember this device too. After that relogin into messenger on Frost. Should be fine.

That did the trick, thanks!

**Comment by kadogo** [comment]
The version https://github.com/AllanWang/Frost-for-Facebook-APK-Builder/releases/tag/v2886 worked for me too. It's a bit slow and less ergnomic but it works and that's already very good ^^

**Comment by AllanWang** [comment]
I don't think speed is going to get any better, and I'd be hesitant to experiment with blocking certain scripts as they might be necessary. There are things you can do such as hide syncing for unread markers, but that would take a lot of testing.

Goal is to fix some immediate problems and try to ship something this weekend for the F-Droid audience

**Comment by AllanWang** [comment]
Initial support is merged into release. Please create new specific issues if any arise, and look out for duplicates

**Comment by KopiasCsaba** [comment]
> Initial support is merged into release. Please create new specific issues if any arise, and look out for duplicates

Thank you very much for your work!

**Comment by alekksander** [comment]
> Initial support is merged into release. Please create new specific issues if any arise, and look out for duplicates

Frost won't load messenger if opening in external apps is enabled — still stands

**Comment by jt196** [comment]
@AllanWang new user here - broken link showing up with basically all of your new builds - 2886, 2894 and 2902 all tried and they're not running the messenger. Any ideas?

**Comment by PrawyCoD1** [comment]
2023 and still doesn't work for me
